pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -325,26 +252,27 @@ func (am *Alertmanager) SaveAndApplyConfig(ctx context.Context, cfg *apimodels.P
 		return fmt.Errorf("failed to serialize to the Alertmanager configuration: %w", err)
 	}
 
-	am.reloadConfigMtx.Lock()
```

Comment:
The diff here is ruined because we need to take Base's lock, which causes nesting - there's no functional change here. We just need to make sure that we lock (with the same lock) all the way down to calling applyConfig which does not use Base's lock.